src/olympia/reviewers/views.py

```diff
@@ -1428,12 +1434,26 @@ def get_object(self, **kwargs):
         self._verify_object_permissions(obj, obj.version)
         return obj
 
+    def get_addon_object(self):
+        if not hasattr(self, 'addon_object'):
+            self.addon_object = get_object_or_404(
+                # The serializer will not need to return much info about the
+                # addon, so we can use just the translations transformer and
+                # avoid the rest.
+                Addon.objects.get_queryset().only_translations(),
+                pk=self.kwargs['addon_pk'])
+        return self.addon_object
```

Comment:
Isn't that kinda what a `cached_property` is for? We aren't re-using the `self.addon_object` and `self.version_object` in any way at this moment unless I'm missing something?

Reply:
I didn't want to change the semantics, it's a method and not a property and that's why I'm re-implementing a caching mechanism instead of using `@cached_property` (The fact that it's a method aligns well with `get_object()` and our other APIs).

We do re-use the version, and that was one reason we had duplicate queries: `get_version_object()` is called in `get_queryset()`, `get_extra_comment_data()` and also in `get_serializer_context()`. The other reason was that the version object queryset was not "starting" from the addon, so every call to `version.addon` later made some additional queries to load the add-on (and its transformers) instead of loading it once. Technically I didn't have to extract the `get_addon_object()` and cache it into a property, but I did for consistency.

Reply:
Right, I forgot where `get_version_object` is called. Works for me, thanks for clarifying.